### PR TITLE
feat: Concat does not generate mutant when both operands are the same

### LINKED ISSUE #1601: Empty diff - Concat swap
| Question    | Answer
| ------------| ---------------
| Infection version | 0.25.3
| Test Framework version | PHPUnit
| PHP version | 8.0
| Platform    | -
| Github Repo | -

Related to #1524 

https://infection-php.dev/r/lpdl

The same issue as described in #1524, it was closed, but not resolved.
Can we somehow skip this mutation, or I need to change my code (for example using str_repeat...) to pass the infection?

Thanks

### PR DESCRIPTION
**Pull request by michalbundyra**
Closes #1601

This PR:

- [x] Adds new feature - see #1601 
- [x] Covered by tests
- [x] Doc PR: not needed

### CODE REVIEW

**Review by maks-rafalko — APPROVED** [review]
Looks good.

IMO psalm's issues can be manually ignored in this case. 

Also, probably it will be safer to do

```diff
- if ($printer->prettyPrint([$node]) !== $printer->prettyPrint([$newNode])) {
+ if ($printer->prettyPrint([clone $node]) !== $printer->prettyPrint([$newNode])) {
```

since `$node` potentially can be modified?

**Comment by maks-rafalko** [comment]
could you please ignore [these Psalm errors](https://github.com/infection/infection/runs/4059304246?check_suite_focus=true#step:7:23)? and we are good to merge

**Comment by michalbundyra** [comment]
@maks-rafalko done

**Comment by michalbundyra** [comment]
Not sure how it should be done then...

with
```php
/** @psalm-suppress ImpureMethodCall */
```

issues were suppressed, but CS failed... 

with:

```php
/* @psalm-suppress ImpureMethodCall */
```

cs passes, but psalm issues are not suppressed....

File: src/Mutator/Operator/Concat.php [hunk 1]
```diff
@@ -93,7 +93,7 @@ public function mutate(Node $node): iterable
 
         $newNode = new Node\Expr\BinaryOp\Concat($left, $right);
 
-        /** @psalm-suppress ImpureMethodCall */
```
Comment (maks-rafalko) [inline src/Mutator/Operator/Concat.php]: it was a correct way. You can check it locally with `make psalm`

Comment (michalbundyra) [reply src/Mutator/Operator/Concat.php]: yeah, I did, it was right, but CS failed :(

Comment (michalbundyra) [reply src/Mutator/Operator/Concat.php]: see: https://github.com/infection/infection/runs/4059480272?check_suite_focus=true

**Comment by maks-rafalko** [comment]
then let's do it in `psalm.xml`:

```diff
<?xml version="1.0"?>
<psalm
    errorLevel="8"
    resolveFromConfigFile="true"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xmlns="https://getpsalm.org/schema/config"
    xsi:schemaLocation="https://getpsalm.org/schema/config vendor/vimeo/psalm/config.xsd"
    errorBaseline="psalm-baseline.xml"
>
    <projectFiles>
        <directory name="src/Mutator" />
        <ignoreFiles>
            <directory name="vendor" />
           <directory name="src/PhpParser" />
        </ignoreFiles>
    </projectFiles>

+    <issueHandlers>
+        <ImpureMethodCall>
+            <errorLevel type="suppress">
+                <file name="src/Mutator/Operator/Concat.php"></file>
+            </errorLevel>
+        </ImpureMethodCall>
+    </issueHandlers>
</psalm>
```

**Comment by maks-rafalko** [comment]
Thank you @michalbundyra

**Pull request merged**